### PR TITLE
[WIP]Dominator rework, gang base forcefields!

### PR DESCRIPTION
**Pull request by silicons**
Does what @KorPhaeron said in https://tgstation13.org/phpBB/viewtopic.php?f=9&t=11130

Also contains some improvements to fields and allows them to block explosions/atmos,generic field directions,fixes,w/e

### CODE REVIEW

File: code/game/gamemodes/gang/dominator.dm [hunk 1]
```diff
@@ -196,7 +231,7 @@
 		priority_announce("Network breach detected in [locname]. The [gang.name] Gang is attempting to seize control of the station!","Network Alert")
 		gang.domination()
 		SSshuttle.registerHostileEnvironment(src)
-		src.name = "[gang.name] Gang [src.name]"
+		src.name = "[gang.name] Gang [name]"
```
Comment (lzimann) [inline code/game/gamemodes/gang/dominator.dm]: \>removed a src in the string but left in the variable

Comment (kevinz000) [inline code/game/gamemodes/gang/dominator.dm]: wew

File: code/game/turfs/turf.dm [hunk 2]
```diff
@@ -61,6 +61,14 @@
 /turf/proc/Initalize_Atmos(times_fired)
 	CalculateAdjacentTurfs()
 
+/turf/get_explosion_block()
+	var/ret = 0
```
Comment (ChangelingRain) [inline code/game/turfs/turf.dm]: . = 0

File: code/game/turfs/turf.dm [hunk 3]
```diff
@@ -61,6 +61,14 @@
...
+	if(density)
+		ret = explosion_block
```
Comment (ChangelingRain) [inline code/game/turfs/turf.dm]: . = explosion_block

File: code/game/turfs/turf.dm [hunk 4]
```diff
@@ -61,6 +61,14 @@
...
+	for(var/atom/A in src)
+		ret += A.get_explosion_block()
```
Comment (ChangelingRain) [inline code/game/turfs/turf.dm]: . += A.get_explosion_block()

File: code/game/turfs/turf.dm [hunk 5]
```diff
@@ -61,6 +61,14 @@
...
+	return ret
```
Comment (ChangelingRain) [inline code/game/turfs/turf.dm]: remove

**Comment by Jordie0608** [comment]
If this is disabled and only to be used with Kor's other ideas, why even PR this? There's not much point to merging it.

**Comment by kevinz000** [comment]
@Jordie0608 it can be used on its own if it's enabled, it means gangs can drop a place to use as a base that will be protected as long as they don't start the domination.

**Comment by kevinz000** [comment]
right i might as well do the entire thing.

**Comment by Jordie0608** [comment]
Well yes; but it isn't enabled.

**Comment by PKPenguin321** [comment]
Not sure I like this still, feels like doms would just get camped out as soon as the crew finds them

**Comment by kevinz000** [comment]
@PKPenguin321 I could make it have a extended range wall to make it less campable
I dunno I'll think of something after I refactor income and talk with kor, I do understand this idea is experimental.

**Comment by kevinz000** [comment]
right i'm not done!

**Comment by kevinz000** [comment]
IT'S BACK ON!

**Comment by Robustin** [comment]
I need more information, stuff like "killing off gangtools" is concerning to me (yes I know you moved influence to the mind).

I feel that gangs need a certain degree of independence from their base and its no fun having to constantly backtrack into a department you probably dont have access to just so you can get another mag of ammo.

I also have no idea how you're going to incorporate this into security-mode. Seems like it would be super easy to find the dom and then sec seals off the area with a shitton of walls and arrests anyone trying to get back in.

**Comment by kevinz000** [comment]
ask @KorPhaeron

**Comment by Cyberboss** [comment]
@kevinz000

**Comment by kevinz000** [comment]
u_u

**Comment by KorPhaeron** [comment]
stupit kevinz

**Comment by kevinz000** [comment]
ok fine il finish it

**Comment by PKPenguin321** [comment]
I still think this is bad, sec will just find the dom 5 seconds in and set up camp around it

**Comment by KorPhaeron** [comment]
Sec wont exist though

**Comment by PKPenguin321** [comment]
Sec, literally anybody that's not part of the gang, same thing

**Comment by KorPhaeron** [comment]
I dont think vigilantes will be coordinated enough to cover every possible exit regularly, and the defenders will have a large advantage being able to enter/exit from the field while the attackers are standing around in a hallway

**Comment by lzimann** [comment]
is this going anywhere @kevinz000

**Comment by kevinz000** [comment]
@KorPhaeron i'll finish this after rnd pr

**Pull request closed without merging**